crates/biome_dependency_graph/src/dependency_graph.rs

```diff
@@ -187,6 +209,60 @@ impl DependencyGraph {
     }
 }
 
+/// Information tracked for every export.
+///
+/// Exports come in three varieties: "own" exports that are defined in the
+/// module itself, re-exports for named exports, and re-exports that apply to
+/// all symbols from another module.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Export {
+    /// An export that is defined in this module.
+    Own(OwnExport),
+    /// An export that is re-exported by this module, but which is defined
+    /// within another module.
+    ///
+    /// E.g. `export { someSymbol } from "other-module"`.
+    Reexport(Import),
+    /// An export that creates an alias for all symbols from another module.
+    ///
+    /// E.g. `export { * as alias } from "other-module"`.
+    ReexportAll(ReexportAll),
+}
+
+/// Information tracked for every "own" export.
+#[derive(Clone, Debug, PartialEq)]
+pub struct OwnExport {
+    /// Optional JSDoc comment associated with the export.
+    ///
+    /// The comment is trimmed and normalized to remove the trivia of the
+    /// comment.
+    ///
+    /// ## Example
+    ///
+    /// Assuming the following export:
+    ///
+    /// ```ts
+    /// /**
+    ///  * Magic constant of fooness.
+    ///  *
+    ///  * For if you want more ways to write 1.
+    ///  */
+    /// export const FOO = 1;
+    /// ```
+    ///
+    /// The comment would be:
+    /// `"Magic constant of fooness.\n\nFor if you want more ways to write 1."`.
+    pub jsdoc_comment: Option<String>,
```

Comment:
nit: I would call it `block_comment` because at this point of the code we don't know if it's a JsDoc

Reply:
See comment above. Block comments that don't start with `/**` are ignored.

Just to be absolutely sure about this, I also looked it up here: https://jsdoc.app/about-getting-started

> JSDoc comments should generally be placed immediately before the code being documented. Each comment must start with a `/**` sequence in order to be recognized by the JSDoc parser. Comments beginning with `/*`, `/***`, or more than 3 stars will be ignored. This is a feature to allow you to suppress parsing of comment blocks.

I'll also update the code to make sure `/***` and more stars will be ignored :)

Reply:
Ok, that makes sense now. Thank you for looking it up